task/upload-sbom-to-trustification/0.1/upload-sbom-to-trustification.yaml

```diff
@@ -0,0 +1,230 @@
+
+        # Symlink the discovered SBOMS to ${WORKDIR}/${sbom_id}.json so that subsequent steps
+        # don't have to look for them again.
+        sbom_path="$WORKDIR/$sbom_id.json"
+        ln -s "$(realpath "$filepath")" "$sbom_path"
+
+        if [[ -n "$supported_version" ]]; then
+          sbom_version="$(jq -r ".specVersion" "$sbom_path")"
+
+          if version_lesser_equal "$sbom_version" "$supported_version"; then
+            echo "SBOM version ($sbom_version) is supported (<= $supported_version), will not convert"
+          else
+            echo "SBOM version ($sbom_version) is not supported, will convert to $supported_version"
+            printf "%s" "$supported_version" > "${sbom_path}.convert_to_version"
+          fi
+        fi
+      done
+
+      echo "Found $(find "$WORKDIR" -name "*.json" | wc -l) CycloneDX SBOMs"
+
+  # Needs syft, which is in a different image => has to be a separate step
+  - name: convert-sboms-if-needed
+    image: registry.redhat.io/rh-syft-tech-preview/syft-rhel9:1.0.1@sha256:27c268d678103a27b6964c2cd5169040941b7304d0078f9727789ffb8ffba370
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit -o nounset -o pipefail
+
+      # Return zero matches when a glob doesn't match rather than returning the glob itself
+      shopt -s nullglob
+
+      for sbom_path in "$WORKDIR"/*.json; do
+        conversion_attr="${sbom_path}.convert_to_version"
+
+        if [[ -f "$conversion_attr" ]]; then
+          cdx_version="$(cat "$conversion_attr")"
+          original_sbom_path="$(realpath "$sbom_path")"
+          original_sbom_relpath="$(realpath "$sbom_path" --relative-base="$SBOMS_DIR")"
+
+          echo "Converting $original_sbom_relpath to CycloneDX $cdx_version"
+          syft convert "$original_sbom_path" -o "cyclonedx-json@${cdx_version}=${sbom_path}.supported_version"
+        else
+          # Just duplicate the symlink, the original SBOM already has a supported CDX version
+          cp --no-dereference "$sbom_path" "${sbom_path}.supported_version"
+        fi
+      done
+
+  - name: upload-sboms
+    image: quay.io/redhat-appstudio/appstudio-utils:5bd7d6cb0b17f9f2eab043a8ad16ba3d90551bc2@sha256:8c7fcf86af40c71aeb58e4279625c8308af5144e2f6b8e28b0ec7e795260e5f7
+    script: |
+      #!/usr/bin/env bash
+      set -o errexit -o nounset -o pipefail
+
+      shopt -s nullglob
+      sboms_to_upload=("$WORKDIR"/*.json)
+
+      if [[ "${#sboms_to_upload[@]}" -eq 0 ]]; then
+        echo "No SBOMs to upload"
+        exit 0
+      fi
+
+      read_required_secret_key() {
+        local key="$1"
+        if [[ -f "$TRUSTIFICATION_SECRET_PATH/$key" ]]; then
+          cat "$TRUSTIFICATION_SECRET_PATH/$key"
+        else
+          echo "Missing configuration: $key" >&2
+          echo "Does the '$TRUSTIFICATION_SECRET_NAME' secret exist in your namespace and contain the required keys?" >&2
+          echo "Refer to the description of this Task for details." >&2
+          exit 1
```

Comment:
@jduimovich does the task need an option to exit with success when TPA is not configured?

Reply:
Gonna merge this, LMK if you find you need this

Reply:
Added it in the end